src/jsdoc_transformer.ts

```diff
@@ -130,29 +130,30 @@ export function maybeAddHeritageClauses(
       return null;
     }
 
-    // Resolve any aliases to the underlying type.
-    if (sym.flags & ts.SymbolFlags.TypeAlias) {
-      // It's implementing a type alias.  Follow the type alias back
-      // to the original symbol to check whether it's a type or a value.
-      const type = mtt.typeChecker.getDeclaredTypeOfSymbol(sym);
-      if (!type.symbol) {
-        // It's not clear when this can happen.
-        mtt.debugWarn(decl, `could not get type of symbol: ${expr.getText()}`);
-        return null;
-      }
-      sym = type.symbol;
-    }
-    if (sym.flags & ts.SymbolFlags.Alias) {
-      sym = mtt.typeChecker.getAliasedSymbol(sym);
+    // localSym can be an alias (e.g. an imported symbol); dealias it before
+    // examining flags, like whether it's a type or a value.  But keep the
+    // original symbol around, because we want to only reference the local
+    // name in any emitted reference.
+    let trueSym = localSym;
```

Comment:
maybe `dereferencedSym`? The other isn't untrue, it's just the local one. Or maybe `declarationSym`, given it's the symbol at the declaration location?